File: Heaps/215. Kth Largest Element in an Array/215. Kth Largest Element in an Array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findKthLargest = function (nums, k) {
  nums = new MaxPriorityQueue();

  for (let num of nums) maxHeap.enqueue(num);

  for (let i = 0; i < k - 1; i++) maxHeap.dequeue();

  return maxHeap.front();
};

// Time: O(n + k log n)
// Space: O(n)

File: Heaps/215. Kth Largest Element in an Array/215. Kth Largest Element in an Array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        for i in range(len(nums)):
            nums[i] *= -1

        heapq.heapify(nums)

        for _ in range(k - 1):
            heapq.heappop(nums)

        return -nums[0]

# Time: O(n + k log n)
# Space: O(1)
